pkg/apiserver/debugapi/service.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debugapi
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/joomcode/errorx"
+
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/debugapi/schema"
+	"github.com/pingcap/tidb-dashboard/pkg/apiserver/user"
+)
+
+var (
+	ErrNS      = errorx.NewNamespace("error.api.debugapi")
+	ErrFromDst = ErrNS.NewType("err_from_dst")
+)
+
+func registerRouter(r *gin.RouterGroup, auth *user.AuthService, s *Service) {
+	endpoint := r.Group("/debugapi")
+	endpoint.Use(auth.MWAuthRequired())
+
+	endpoint.GET("/proxy", s.Proxy)
```

Comment:
I'm confused about whether the proxy service should be placed in the gin routes. Is this a good practice?